Created page.tsx

Dhruvilp25 opened the pull request:
Starting changelogs page 
issue #132

Dhruvilp25 committed:
Create page.tsx

Starting changelogs page

jayptz commented:
nah fam chalk this developer, ggs bro alr failed.

jayptz reviewed (changes requested):
buns mcgee no changes

devp19 commented:
/page.tsx is blank which is causing error. make sure to commit changes with the file and then do another pr under same branch

jayptz commented:
> /page.tsx is blank which is causing error. make sure to commit changes with the file and then do another pr under same branch

He was just testing pull requests and how to do it that's why page.tsx was blank @devp19

Dhruvilp25 committed:
added the changelogs template (basic features so far)

jayptz reviewed (changes requested):
### Changelog: Backfill pre-June entries, switch to [GitHub-style changelog UI](https://github.blog/changelog/) (quarterly), and add Navbar

**Summary**  
The current changelog is clean but incomplete. We need to backfill pre-June content, and restructure the UI to use **quarterly groupings** (like [GitHub’s changelogs](https://github.blog/changelog/)). Also, add the Navbar for consistency with the landing page.  

---

#### Must-do
- [ ] **Confirm scope and UI direction with @devp19** 
- [ ] **Request earlier (pre-June) changes from Tirth** and gather entries  
- [ ] **Switch to [GitHub-style changelog](https://github.blog/changelog/) quarterly layout**  
  - Group changes by **quarter** (Q1, Q2, Q3, Q4) with year labels (e.g., `Q2 2025`)  
  - Add tag chips (e.g., `frontend`, `api`, `infra`)
  - Since there will be too much info per quarter, make it so you can just summarize the changelogs for a quick read, and if a user wants to read more, they'll click on said changelog for an in-depth explanation.  
- [ ] **Add Navbar** to the changelog page  
  - Copy from `resdex-v2/src/app/page.tsx` lines **363–453**  
  - Ensure imports/utilities are included  
  - Keep the same spacing, logo sizing, and active link state

jayptz commented:
Other than that i mess with it

devp19 reviewed (changes requested):
this is good starting, much better than i expected. refer to jays comment, add same navbar, (just copy it from the /digest page.tsx since it uses the new version. 

Also, can we add a vertical thin line in between the changelog entries?

--> change folder name to Changelog instead of plural Changelogs (you may have to update other content pages if you reference imports that way. 

--> Change date to be September 1st, 2025 format instead of 7/1/2025

--> refer to image below, can we change the rightside layout like that? hero image and then explanations. you can copy it exactly. 

<img width="1099" height="733" alt="Screenshot 2025-09-05 at 11 07 09 AM" src="https://github.com/user-attachments/assets/725be675-ce55-4cfe-83e4-dc66be43964e" />

devp19 commented on resdex-v2/src/app/changelogs/content/2025-04-22.mdx:
```diff
@@ -0,0 +1,19 @@
+---
+title: "Bug fixes and polish"
+description: "Resolved several UI bugs and fixed broken links."
+date: "2025-04-22"
```
change all the dates in .mdx files to be actual long form date. can keep the file name as short form though if that helps.

Dhruvilp25 committed:
Changelogs page

Changed up the UI, search bar added (prolly more useful with more logs), filters for tags and date, added the Quarters and grouped them by it, dropdowns to show more info

Dhruvilp25 commented:
Ok so I still gotta get the entries from tirth but I wanted to get the UI down first, I changed it up and I tried using images but it didn't look right and I thought this looked fine, if you still want them lmk Ill change it around and make it work. Then I will add all our entries and sort everything. Also lmk what else you wanna remove or add. :)

devp19 commented:
not bad, lower fontsize and logo a bit for title header. make it so that I can click the box to expand it down instead of having to click the actual dropdown arrow on top right corner. (keep both tho).

jayptz reviewed (changes requested):
Please make the arrows for the dropdown at the bottom of the box instead of the top right.

As to what @devp19  said, "click the box to expand it down instead of having to click the actual dropdown arrow," add this, but just make the arrow bottom-middle

Also, add "changelogs" link to the bottom of the webpage under "Company" so you don't have to do /changelogs every time.


Nice tags, love the filter box. If possible, add a "year" filter to it as well instead of the data range. Make it so the user can just select which year. 

Other than that, nice work!
Check in with @devp19 before implementing my changes.

Dhruvilp25 committed:
Changelog more fixes

I have updated the header size and changed the boxes to be clickable, added the changelogs links, and got rid of the year range for a better year and quarter filters. I tried out the arrow in the bottom middle but it looks off as the right side is too empty, if yall wanna display any other info I can put it there and move the arrow.

jayptz reviewed (commented):
Nice! I'm assuming now you're just waiting for Tirth's Changelogs. Love the UI, ngl it's chill to keep it like this until the TIrth sends the new logs no? @devp19  put him onto the next issue?

nice i see taht you also changed the filtering to years.

jayptz reviewed (commented):
added the Changelog under company, nice i just peeped that.

devp19 commented:
@Dhruvilp25 can u commit any changes from prev update? mentioned you got the actual content within each changelog entry. want to see how it looks like populated.

Dhruvilp25 committed:
Added all the logs that we currently have

devp19 reviewed (changes requested):
<img width="1326" height="416" alt="Screenshot 2025-09-27 at 3 52 13 PM" src="https://github.com/user-attachments/assets/34aab130-da46-4402-8532-90504d4e69c7" />

Can we not have to do drop downs? Just show as is. Also can we expand a bit more on the actual entries instead of basic fixes? Like a small 1-2 sentence description

Dhruvilp25 committed:
Added more info to each entry

For now I  js added more to each entry. Also Dev what dropdowns do you want me to remove the main drop down so features and bug fixes always show but have drop downs, or the features and bug fixes drop down but keep the main one or even all of them. However I feel like u should keep it cuz it makes all the changelogs concise and people can open up what they wanna know otherwise it might be too much on the screen. But lmk what you would like.

devp19 commented:
thanks can we also look into the prev change req 

--> Can we not have to do drop downs? Just show as is. Also can we expand a bit more on the actual entries instead of basic fixes? Like a small 1-2 sentence description
